pkg/versioning/helm.go

```diff
@@ -0,0 +1,96 @@
+package versioning
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+// JSONfile defines an artifact using a json file for versioning
+type HelmChart struct {
+	path             string
+	metadata         chart.Metadata
+	utils            Utils
+	updateAppVersion bool
+}
+
+func (h *HelmChart) init() error {
+	if h.utils == nil {
+		return fmt.Errorf("no file utils provided")
+	}
+	if len(h.path) == 0 {
+		charts, err := h.utils.Glob("**/Chart.yaml")
+		if len(charts) == 0 || err != nil {
+			return fmt.Errorf("failed to find a helm chart file")
+		}
+		// use first chart which can be found
+		h.path = charts[0]
+	}
+
+	if len(h.metadata.Version) == 0 {
+		content, err := h.utils.FileRead(h.path)
+		if err != nil {
+			return fmt.Errorf("failed to read file '%v': %w", h.path, err)
+		}
+
+		err = yaml.Unmarshal(content, &h.metadata)
+		if err != nil {
+			return fmt.Errorf("helm chart content invalid '%v': %w", h.path, err)
+		}
+	}
+
+	return nil
+}
+
+// VersioningScheme returns the relevant versioning scheme
+func (h *HelmChart) VersioningScheme() string {
+	return "semver2"
+}
+
+// GetVersion returns the current version of the artifact with a JSON-based build descriptor
+func (h *HelmChart) GetVersion() (string, error) {
+	if err := h.init(); err != nil {
+		return "", fmt.Errorf("failed to init helm chart versioning: %w", err)
+	}
+
+	return h.metadata.Version, nil
+}
+
+// SetVersion updates the version of the artifact with a JSON-based build descriptor
+func (h *HelmChart) SetVersion(version string) error {
+	if err := h.init(); err != nil {
+		return fmt.Errorf("failed to init helm chart versioning: %w", err)
+	}
+
+	h.metadata.Version = version
```

Comment:
Don't agree with this one. The helm chart version must not match the artifact version

Reply:
this functionality just sets the value for the version as per SetVersion, e.g. `buildTool: helm`, `GetVersion` will read from chart and `SetVersion` will update ...